packages/electron-updater/src/MacUpdater.ts

```diff
@@ -50,13 +50,18 @@ export class MacUpdater extends AppUpdater {
       log.warn(`sysctl shell command to check for macOS Rosetta environment failed: ${e}`)
     }
 
+    const isArm64Mac = process.arch === "arm64" || isRosetta
+
     // allow arm64 macs to install universal or rosetta2(x64) - https://github.com/electron-userland/electron-builder/pull/5524
     const isArm64 = (file: ResolvedUpdateFileInfo) => file.url.pathname.includes("arm64") || file.info.url?.includes("arm64")
-    if (files.some(isArm64)) {
-      files = files.filter(file => (process.arch === "arm64" || isRosetta) === isArm64(file))
+    if (isArm64Mac && files.some(isArm64)) {
+      files = files.filter(file => isArm64Mac === isArm64(file))
+    } else {
+      files = files.filter(file => !isArm64(file))
```

Comment:
This is actually unnecessary as if there are no ARM64 files, we don't need to filter the files at all?